src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/MultipleReturnPointsInspector.java

```diff
@@ -0,0 +1,83 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeSmell;
+
+/*
+ * This file is part of the Php Inspections (EA Extended) package.
+ *
+ * (c) Funivan <alotofall@gmail.com>
+ * (c) Vladimir Reznichenko <kalessil@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiRecursiveElementWalkingVisitor;
+import com.jetbrains.php.lang.psi.elements.Function;
+import com.jetbrains.php.lang.psi.elements.Method;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import com.jetbrains.php.lang.psi.elements.PhpReturn;
+import com.jetbrains.php.lang.psi.visitors.PhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.ExpressionSemanticUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class MultipleReturnPointsInspector extends BasePhpInspection {
+    private static final String message = "Method with multiple return points.";
+
+
+    @NotNull
+    public String getShortName() {
+        return "MultipleReturnPointsInspection";
+    }
+
+    @NotNull
+    public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder holder, boolean isOnTheFly) {
+        return new PhpElementVisitor() {
+
+            public void visitPhpMethod(Method method) {
+                final PhpClass containingClass = method.getContainingClass();
+                final PsiElement nameIdentifier = method.getNameIdentifier();
+                if (containingClass != null && nameIdentifier != null && !containingClass.isTrait()) {
+
+                    final ReturnPointCountVisitor countVisitor = new ReturnPointCountVisitor(method);
+                    method.accept(countVisitor);
```

Comment:
@artspb, requesting for support =) 

What are advantages of using visitors vs PsiTreeUtil search methods+regular foreach loop? Are there any, and why if any?